pkg/machine/applehv/machine.go

```diff
@@ -113,87 +99,76 @@ func (m *MacMachine) Init(opts machine.InitOptions) (bool, error) {
 		m.ImageStream = "custom"
 		g, err := machine.NewGenericDownloader(vmtype, m.Name, opts.ImagePath)
 		if err != nil {
-			return false, err
+			return err
 		}
 		imagePath, err := machine.NewMachineFile(g.Get().LocalUncompressedFile, nil)
 		if err != nil {
-			return false, err
+			return err
 		}
 		m.ImagePath = *imagePath
 		if err := machine.DownloadImage(g); err != nil {
-			return false, err
+			return err
 		}
 	}
+	return nil
+}
 
-	logPath, err := machine.NewMachineFile(filepath.Join(dataDir, fmt.Sprintf("%s.log", m.Name)), nil)
-	if err != nil {
-		return false, err
-	}
-	m.LogPath = *logPath
-	runtimeDir, err := m.getRuntimeDir()
-	if err != nil {
-		return false, err
-	}
-
-	readySocket, err := machine.NewMachineFile(filepath.Join(runtimeDir, fmt.Sprintf("%s_ready.sock", m.Name)), nil)
-	if err != nil {
-		return false, err
-	}
-
+func setGVProxyInfo(m *MacMachine, runtimeDir string) error {
```

Comment:
Can't we make this a method of the MacMachine?

```func (m *MacMachine) setGVProxyInfo(runtimeDir string) error {```

Reply:
Should all be fixed.